### PR TITLE
feat: add C implementation for `math/base/special/minmax`

### LINKED ISSUE #2106: [RFC]: Add C implementation for `@stdlib/math/base/special/minmax`
### Description

This RFC proposes to add a C implementation for [`@stdlib/math/base/special/minmax`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/math/base/special/minmax)

### Related Issues

Progresses #649 

### Questions

- I note that the implementations of `min` and `max` are already available. Should the C implementation exactly follow the logic in the JS files, or could it use the `min` and `max` functions in C while maintaining the same function signature and return values?

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Userf1995**
Resolves #2106  .

## Description

> What is the purpose of this pull request?

This pull request:

-   add C implementation for math/base/special/minmax

## Related Issues

> Does this pull request have any related issues?

This pull request:

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [X] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/minmax/README.md [hunk 1]
```diff
@@ -119,6 +119,100 @@ for ( i = 0; i < 100; i++ ) {
...
+#### stdlib_base_minmax( x, y, min, max )
+
+Returns the minimum and maximum values in a single pass.
+
+```c
+double var v = stdlib_base_minmax( 4.2, 3.14 );
+// returns [ 3.14, 4.2 ]
+
+double var v = stdlib_base_minmax( +0.0, -0.0 );
+// returns [ -0.0, +0.0 ];
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/minmax/README.md]: Example code needs to be updated as it currently doesn't match function signature above.

File: lib/node_modules/@stdlib/math/base/special/minmax/src/main.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+*
+* @example
+* double v = stdlib_base_minmax( 3.14, NaN );
+* // returns [ NaN, NaN ]
+*
+* @example
+* double v = stdlib_base_minmax( +0.0, -0.0 );
+* // returns [ -0.0, 0.0 ]
+*/
+double stdlib_base_minmax( const double x, const double y, double* min, double* max ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/minmax/src/main.c]:
```suggestion
void stdlib_base_minmax( const double x, const double y, double* min, double* max ) {
```

File: lib/node_modules/@stdlib/math/base/special/minmax/src/main.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+#include "stdlib/math/base/special/minmax.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/assert/is_negative_zero.h"
+
+/**
+* Returns the minimum and maximum values.
+*
+* @param {number} x - first number
+* @param {number} y - second number
+* @returns {Array<number>} minimum and maximum values
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/minmax/src/main.c]: Need to update the parameter and return annotations here. Notice this is not JSDoc.

File: lib/node_modules/@stdlib/math/base/special/minmax/test/test.native..js [hunk 4]
```diff
@@ -0,0 +1,129 @@
+/**
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/minmax/test/test.native..js]: File has two periods right now, please change `..js` to `.js`.

File: lib/node_modules/@stdlib/math/base/special/minmax/src/main.c [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+*
+* @example
+* double v = stdlib_base_minmax( 3.14, 4.2 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/minmax/src/main.c]: Examples need to be updated to match implementation.

File: lib/node_modules/@stdlib/math/base/special/minmax/src/addon.c [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+/**
+* Receives JavaScript callback invocation data.
+*
+* @private
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/minmax/src/addon.c:32]: Should be possible to significantly simplify this file via the utilities in https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/napi.

File: lib/node_modules/@stdlib/math/base/special/minmax/include/stdlib/math/base/special/minmax.h [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Returns the minimum and maximum values.
+*/
+double stdlib_base_minmax( const double x, const double y, double *min, double *max );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/minmax/include/stdlib/math/base/special/minmax.h]:
```suggestion
void stdlib_base_minmax( const double x, const double y, double *min, double *max );
```

File: lib/node_modules/@stdlib/math/base/special/minmax/benchmark/c/native/benchmark.c [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+*
+* @return elapsed time in seconds
+*/
+double benchmark() {
+	double min;
+	double elapsed;
+	double t;
+	double x;
+	double y;
+	double z;
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/minmax/benchmark/c/native/benchmark.c]: Would be good to rename `z` to `max` here.

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Thanks for this PR! Left some comments that will need to be addressed.

File: lib/node_modules/@stdlib/math/base/special/minmax/src/main.c [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+	return;
+    }
+    if ( x < y ) {
+        *min = x;
+	*max = y;
+	return;
+    }
+    *min = y;
+    *max = x;
+    return;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/minmax/src/main.c:61]: You have mixed tab and space indentation. It should be TAB indentation.

**Pull request closed without merging**